tests/pass/shims/windows-rand.rs

```diff
@@ -0,0 +1,41 @@
+//@only-target-windows: this directly tests windows only random functions
+use core::ffi::c_void;
+use core::mem::size_of_val;
+use core::ptr::null_mut;
+
+fn main() {
+    let mut key = [0u8; 24];
+    let len: u32 = size_of_val(&key).try_into().unwrap();
+    let ret = unsafe {
+        BCryptGenRandom(null_mut(), key.as_mut_ptr(), len, BCRYPT_USE_SYSTEM_PREFERRED_RNG)
+    };
+    // NTSTATUS codes use the high bit to indicate an error
+    assert!(ret >= 0);
+
+    let ret = unsafe { BCryptGenRandom(BCRYPT_RNG_ALG_HANDLE, key.as_mut_ptr(), len, 0) };
+    assert!(ret >= 0);
+
+    let ret = unsafe { RtlGenRandom(key.as_mut_ptr(), len) };
+    // RtlGenRandom returns a BOOLEAN where 0 indicates an error
+    assert_ne!(ret, 0);
+}
+
+// Windows API definitions.
```

Comment:
Can you put the definitions above the code where they are used?

Other than that, looks good to me, thanks. :)

Reply:
Done.